libbeat/api/npipe/listerner_posix.go

```diff
@@ -0,0 +1,33 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//+build !windows
+
+package npipe
+
+import (
+	"context"
+	"errors"
+	"net"
+)
+
+// DialContext create a Dial to be use with an http.Client to connect to a pipe.
+func DialContext(npipe string) func(context.Context, string, string) (net.Conn, error) {
+	return func(ctx context.Context, _, _ string) (net.Conn, error) {
+		return nil, errors.New("named pipe doesn't work on linux")
```

Comment:
Doesn't linux support named pipes using `mkfifo`?

Reply:
They are supported through `syscall.mkfifo` and I think we could support them but that would require a bit of close to the metal implementation and should be similar to what go-winio is doing in https://github.com/microsoft/go-winio/blob/84b4ab48a50763fe7b3abcef38e5205c12027fac/pipe.go#L185-L203. Since they aren't working out of the box, and I don't know all the gotcha for the implementation I consider them not required for the first version of this implementation.

Reply:
I'm okay with postponing mkfifo.

Consider error wording: "names pipes are only supported on Windows".

Reply:
Thanks for the explanation, @ph. I'll mark this as resolved.